core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategy.java

```diff
@@ -66,21 +70,22 @@ public boolean shouldRetry(I request, Throwable throwable) {
     }
 
     /**
-     * Returns {@link CompletableFuture} that contains a {@link Boolean} flag which indicates whether
-     * {@link RetryingClient} needs to retry or not.
-     * <p>
-     * If an {@link Exception} occurs while processing {@link Request} and {@link Response}, this method
+     * Returns {@link CompletableFuture} with {@link Optional} that contains {@link Backoff} which will be
+     * used for retry. If the condition does not match, this needs to return {@link Optional#EMPTY}
+     * to stop retry attempt.
+     *
+     * <p>If an {@link Exception} occurs while processing {@link Request} and {@link Response}, this method
      * should not complete the {@link CompletableFuture} with the {@link Exception}. The {@link Exception}
-     * needs to be dealt in the {@link RetryStrategy#shouldRetry(Request, Throwable)} method, and
-     * the {@link CompletableFuture} has to be completed with {@code false}.
-     * </p>
+     * needs to be dealt in the {@link RetryStrategy#shouldRetry(Request, Throwable)} method, therefore
+     * the {@link CompletableFuture} has to be completed with {@link Optional#EMPTY}.
      */
-    CompletableFuture<Boolean> shouldRetry(I request, O response);
+    CompletableFuture<Optional<Backoff>> shouldRetry(I request, O response);
```

Comment:
@imasahiro Sorry, I have to change the api again in order to meet the needs in #655